### PR TITLE
[ALP-85] PUT /volunteers working

### LINKED ISSUE #85: [ALP 85] Create /volunteers [PUT] endpoint and implement 'clear' functionality

### PR DESCRIPTION
**Pull request by michaelfromyeg**
Updated getVolunteers and getVolunteerNames to check for isDeleted, added query on button press, table refreshes after button press.

### CODE REVIEW

File: aunt-leahs-app/src/components/CustomTable.jsx [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
 
 // REQUIRES: data prop has to be array of objects where each object key represents the name of the field in camelcase.
 //					 Refer to data in constants.jsx
 //					 each field in data should be either string or boolean. boolean fields are displayed '✓' or '✗'
 
-const formatHeaderTitle = (title) => {
-	return title.replace(/([A-Z])/g, ' $1').replace(/^./, (str) => str.toUpperCase());
+const formatHeaderTitle = title => {
+  return title
+    .replace(/([A-Z])/g, " $1")
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/CustomTable.jsx]: What does this do?

File: aunt-leahs-app/src/components/CustomTable.jsx [hunk 2]
```diff
@@ -1,52 +1,62 @@
...
+    .replace(/^./, str => str.toUpperCase());
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/CustomTable.jsx]: This entire line can be replaced with `.toUpperCase()`, I think

File: aunt-leahs-app/src/components/CustomTable.jsx [hunk 3]
```diff
@@ -1,52 +1,62 @@
...
+              <TableRow key={row.id}>
+                {Object.keys(row)
+                  .filter(key => key !== "id")
+                  .map((key, idx) => (
+                    <TableCell className="tableCell" align="left" key={idx}>
+                      {typeof row[key] == "boolean"
+                        ? row[key]
+                          ? "Yes"
+                          : "No"
+                        : row[key]}
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/CustomTable.jsx]: The logic should be put into a function, imo

File: aunt-leahs-functions/shifts/index.js [hunk 4]
```diff
@@ -1,124 +1,123 @@
...
-            context.res = {
-                body: { message: 'Successfully deleted shifts from the database' }
-            };
+    connection.execSql(request);
+  }
 
-            context.done();
-        });
+  function deleteShifts() {
+    var queryString = "UPDATE Shift SET isDeleted = 1;";
```
Comment (justinkkwan) [inline aunt-leahs-functions/shifts/index.js]: Needs `WHERE` condition

File: aunt-leahs-functions/volunteers/index.js [hunk 5]
```diff
@@ -1,130 +1,165 @@
...
-                    context.log(err);
-                    context.done();
-                }
-            });
-
-        connection.execSql(request);
-    }
+    request.on("doneProc", function(rowCount, more, returnStatus, rows) {
+      context.res = {
+        body: JSON.stringify(volunteers)
```
Comment (justinkkwan) [inline aunt-leahs-functions/volunteers/index.js]: I don't think this needs to be stringify'd

File: aunt-leahs-functions/volunteers/index.js [hunk 6]
```diff
@@ -1,130 +1,165 @@
...
+
+  function putVolunteers() {
+    var queryString = `UPDATE Volunteer SET isDeleted = 1;`;
+
+    request = new Request(queryString, function(err) {
+      if (err) {
+        context.log(err);
+
+        context.res = {
+          body: "Error occurred deleting volunteers from the database " + err
```
Comment (justinkkwan) [inline aunt-leahs-functions/volunteers/index.js]: Maybe end the string with `: \n` before the `+  err`?

**Review by justinkkwan — CHANGES_REQUESTED** [review]

**Review by thaingocnguyen — CHANGES_REQUESTED** [review]
There is still a warning when the volunteer data page is loaded due to missing keys. Can you look modify volunteer data is set to fix this?

Other than this, clear functionality is working

File: aunt-leahs-app/src/components/CustomTable.jsx (hunk 1 above)
Comment (michaelfromyeg) [reply aunt-leahs-app/src/components/CustomTable.jsx]: [Looks](https://stackoverflow.com/questions/2418497/javascript-regexp-pattern-matching) like it adds spacing to camelCase

File: aunt-leahs-app/src/components/CustomTable.jsx (hunk 2 above)
Comment (michaelfromyeg) [reply aunt-leahs-app/src/components/CustomTable.jsx]: I think this just grabs the first character (^) and capitalizes it, so it fixes the lowerCaseCase in combination with the last one to "Lower Camel Case".

File: aunt-leahs-app/src/components/CustomTable.jsx (hunk 3 above)
Comment (michaelfromyeg) [reply aunt-leahs-app/src/components/CustomTable.jsx]: Agreed, done!

**Comment by codygagnon** [comment]
Michael, can you please resolve the conflicts?

**Review by justinkkwan — APPROVED** [review]
👍

**Review by codygagnon — APPROVED** [review]

**Review by thaingocnguyen — APPROVED** [review]

**Pull request merged**